lib/arelastic/aggregations/sampler.rb

```diff
@@ -0,0 +1,13 @@
+module Arelastic
+  module Aggregations
+    class Sampler < Bucket
+      def initialize(name, shard_size, aggs)
+        super(name, aggs: aggs, shard_size: shard_size)
```

Comment:
Thoughts on initialize(name, options)?

The read_option! method would be used. 

Or does this signature follow a pattern of another aggregation?

Reply:
I could just drop the constructor and utilize the already existing one in `Arelastic::Aggregations::Bucket`.

I was looking at [`Arelastic::Aggregations::Nested`](https://github.com/matthuhiggins/arelastic/blob/536594cf7aa9de875a5717729543e69079a29a45/lib/arelastic/aggregations/nested.rb) when I made this.

It wasn't clear to me when to use `read_option!` to populate an instance variable and when to use `options` directly.

Reply:
I introduced `read_option!` because kwargs do not support strings, and that sucks in general.

I think your API is fine. `shard_size` is a required param and 3 args is comprehensible.

`read_option!` is used for assigning required options to an accessor. This makes the node a bit more than a container for another `Hash`. I do not have a specific set of logistics here ... all I care about is that API for initializing a node and the output of `as_elastic`.